06_puppy/willshen8/mapStore_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type MapStoreTestSuite struct {
+	suite.Suite
+}
+
+func (suite *MapStoreTestSuite) TestCreateAndReadPuppy() {
+	storer := make(MapStore)
+	newPuppy := Puppy{1, "Jack Russell", "White and Brown", "500"}
+	storer.CreatePuppy(newPuppy)
+	readPuppy := storer.ReadPuppy(newPuppy.ID)
+	suite.Equal(newPuppy, *readPuppy)
+}
+
+func (suite *MapStoreTestSuite) TestReadNonExistentPuppy() {
+	storer := make(MapStore)
+	readPuppy := storer.ReadPuppy(1)
+	suite.Nil(readPuppy)
+}
+
+func (suite *MapStoreTestSuite) TestUpdatePuppy() {
+	var store4 Storer = make(MapStore)
```

Comment:
in my mind name `store` would be bwtter than `store4`. please consider `s/var store[0-9]/var store/g`